src/libcore/cmp.rs

```diff
@@ -95,6 +96,35 @@ use self::Ordering::*;
 /// assert!(b1 != b3);
 /// ```
 ///
+/// ## How can I implement it to compare with another type?
+///
+/// This is pretty easy to do. Let's take back our previous code and modify it:
```

Comment:
Not sure i agree with using "pretty easy" here. It's a poor phrase to use in documentation. How about something like this:

> The type you can compare with is controlled by `PartialEq`'s type parameter. For example, let's tweak our previous code a bit:

Then you can follow up below the example with something like:

> By changing `impl PartialEq for Book` to `impl PartialEq<BookFormat> for Book`, we've changed what type we can use on the right side of the `==` operator. This lets us use it in the `assert!` statements at the bottom.

It would also be cool if you added one more example, combining them to show off that the same type can `==` to different types. Maybe continue with:

> You can even combine these implementations to compare against both `Book` and `BookFormat`, like this:
>
> (code sample with both `PartialEq` impls and both sets of assertions)